src/server/db_slice.cc

```diff
@@ -1663,9 +1665,10 @@ void DbSlice::PerformDeletionAtomic(Iterator del_it, ExpIterator exp_it, DbTable
 
   ssize_t value_heap_size = pv.MallocUsed(), key_size_used = del_it->first.MallocUsed();
   stats.inline_keys -= del_it->first.IsInline();
-  AccountObjectMemory(del_it.key(), del_it->first.ObjType(), -key_size_used,
-                      table);                                                // Key
-  AccountObjectMemory(del_it.key(), pv.ObjType(), -value_heap_size, table);  // Value
+
+  slot = KeySlotIfNoSlotId(del_it.key(), slot);
```

Comment:
What I don't understand here is why we need `KeySlotIfNoSlotId` at all? The slot that you pass in this function should map to `del_it.key()`?